### PR TITLE
only list iceberg tables for HiveCatalog listTables method

### PR DESCRIPTION
**Pull request by zhangjun0x01**
this pr related to #1830

### CODE REVIEW

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java [hunk 1]
```diff
@@ -118,8 +119,11 @@ public HiveCatalog(
 
     try {
       List<String> tables = clients.run(client -> client.getAllTables(database));
-      List<TableIdentifier> tableIdentifiers = tables.stream()
-          .map(t -> TableIdentifier.of(namespace, t))
+      List<Table> tableObject = clients.run(client -> client.getTableObjectsByName(database, tables));
```
Comment (rdsr) [inline hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java:122]: nit: maybe use `tableObjects` or `tables` for this and use `tableNames` for prev.

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java [hunk 2]
```diff
@@ -118,8 +119,11 @@ public HiveCatalog(
...
+      List<TableIdentifier> tableIdentifiers = tableObject.stream()
+          .filter(table -> BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE
+              .equalsIgnoreCase(table.getParameters().get(BaseMetastoreTableOperations.TABLE_TYPE_PROP)))
```
Comment (rdsr) [inline hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java:125]: in `org.apache.hadoop.hive.metastore.api.Table` class I do see internally to check for `table.getParameters` is null or not. Do you think that check makes sense here? I'm not completely certain whether `table.getParameters` won't be null

File: hive-metastore/src/test/java/org/apache/iceberg/hive/HiveTableTest.java [hunk 3]
```diff
@@ -298,14 +308,43 @@ public void testFailure() throws TException {
...
+    org.apache.hadoop.hive.metastore.api.Table hiveTable = createHiveTable(hiveTableName);
+    metastoreClient.createTable(hiveTable);
+
+    List<TableIdentifier> tableIdents1 = catalog.listTables(TABLE_IDENTIFIER.namespace());
+    Assert.assertEquals("should only 1 iceberg table .", 1, tableIdents1.size());
+    Assert.assertTrue(catalog.tableExists(TABLE_IDENTIFIER));
+    metastoreClient.dropTable(DB_NAME, hiveTableName);
+  }
+
+  private org.apache.hadoop.hive.metastore.api.Table createHiveTable(String hiveTableName) throws IOException {
```
Comment (rdsr) [inline hive-metastore/src/test/java/org/apache/iceberg/hive/HiveTableTest.java:331]: a simpler way IMO could have been to use HiveCatalog and Iceberg apis  to create a table and then drop the iceberg metadata from Hive table properties. Though I guess this is fine too.

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java (hunk 2 above)
Comment (zhangjun0x01) [reply hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java:125]: Yes,I added the check

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java (hunk 1 above)
Comment (zhangjun0x01) [reply hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java]: I udpated it, it is more clear

File: hive-metastore/src/test/java/org/apache/iceberg/hive/HiveTableTest.java (hunk 3 above)
Comment (zhangjun0x01) [reply hive-metastore/src/test/java/org/apache/iceberg/hive/HiveTableTest.java:331]: I think we can remove the properties from iceberg table, but it is not a correct hive table. Although the unit test can pass, I am not sure whether it makes sense.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Looks good to me. @rdsr, are you okay with merging this?

**Comment by rdsr** [comment]
LGTM! I just merged this!

**Comment by openinx** [comment]
@zhangjun0x01 @rdsr  Thanks for the patch and reviewing work.  It would be better if we can follow the commit message rule that has a `module` name and a message log to explain that's the change.  I mean this PR commit message can be written as `Hive: Only provide iceberg tables in HiveCatalog listTables` .   It's okay for this one if we've really merged this :-).

**Comment by zhangjun0x01** [comment]
> @zhangjun0x01 @rdsr Thanks for the patch and reviewing work. It would be better if we can follow the commit message rule that has a `module` name and a message log to explain that's the change. I mean this PR commit message can be written as `Hive: Only provide iceberg tables in HiveCatalog listTables` . It's okay for this one if we've really merged this :-).

OK,I updated it

**Pull request merged**